releases: add TiDB 4.0.3 release notes (#3291)
pingcap/docs: releases/4.0.3.md

    - Fix the problem of stale scheduler after leader change [#2608](https://github.com/pingcap/pd/pull/2608)
    - Fix the issue that when enable placement rules, sometimes replicas of region may not be able to adjust to the best location [#2605](https://github.com/pingcap/pd/pull/2605)
    - Fix the issue that the deploy path of the store is not updated if the deploy directory is moved [#2600](https://github.com/pingcap/pd/pull/2600)
    - Prevent store limit from changing to zero [#2588](https://github.com/pingcap/pd/pull/2588)
+ Dashboard
    - Fix TiDB connection error when TiDB is scaled-in [#689](https://github.com/pingcap-incubator/tidb-dashboard/pull/689)
    - Fix TiFlash instances not displayed in log searching page [#680](https://github.com/pingcap-incubator/tidb-dashboard/pull/680)
    - Fix metrics selection reset after refresh in overview page [#663](https://github.com/pingcap-incubator/tidb-dashboard/pull/663)
    - Fix connection problem in some TLS scenarios [#660](https://github.com/pingcap-incubator/tidb-dashboard/pull/660)
    - Fix language dropdown not displayed correctly in some case [#677](https://github.com/pingcap-incubator/tidb-dashboard/pull/677)

> + TiFlash

Reviewer: ```suggestion

+ TiFlash
```

    - Fix the issue that TiFlash crashes after renaming primary key column name. [#916](https://github.com/pingcap/tics/pull/916)
    - Fix the issue that concurrent learner read and remove region may lead to deadlocks. [#872](https://github.com/pingcap/tics/pull/872)

+ TiCDC

    - Fix the issue that TiCDC leaks memory in some cases [#704](https://github.com/pingcap/ticdc/pull/704)
    - Fix the issue that unquoted table name causes SQL syntax error [#676](https://github.com/pingcap/ticdc/pull/676)
    - Fix processor doesn't fully exit after `p.stop` is called [#693](https://github.com/pingcap/ticdc/pull/693)

+ BR

    - Fix the problem that backup time may be negative [#405](https://github.com/pingcap/br/pull/405)

+ Dumpling

    - Fix the issue that dumpling omit the NULL value when `--r` is specified [#119](https://github.com/pingcap/dumpling/pull/119)
    - Fix the bug that flush tables may not work for tables to dump [#](https://github.com/pingcap/dumpling/pull/117)

> + TiDB-Lightning

Reviewer: ```suggestion
+ TiDB Lightning
```

    - Active tidb_allow_batch_cop by default. [#18552](https://github.com/pingcap/tidb/pull/18552)
    - Fix issue of query `information_schema.columns` cost too many memory when there are lots of tables in TiDB. [#18436](https://github.com/pingcap/tidb/pull/18436)

+ TiKV

    - A new configuration `backup.num-threads` is introduced to control the backup thread pool size. [#8199](https://github.com/tikv/tikv/pull/8199)
    - Do not send store heartbeat when receiving snapshot [#8136](https://github.com/tikv/tikv/pull/8136)

+ PD

    - Support json formatted log [#2565](https://github.com/pingcap/pd/pull/2565)
+ Dashboard
    - Improved KeyViz bucket merge for cold logical ranges [#674](https://github.com/pingcap-incubator/tidb-dashboard/pull/674)
    - Renamed config item disable-telemetry to enable-telemetry to be more consistent [#684](https://github.com/pingcap-incubator/tidb-dashboard/pull/684)
    - Show progress bar when switching pages [#661](https://github.com/pingcap-incubator/tidb-dashboard/pull/661)
    - Slow log search now follows the same behaviour as log search when there are space separators [#682](https://github.com/pingcap-incubator/tidb-dashboard/pull/682)

> + TiFlash

Reviewer: ```suggestion

+ TiFlash
```

    - Change the unit of the panels of DDL jobs in Grafana to operations per minute [#923](https://github.com/pingcap/tics/pull/923)
    - Reduce IOPS in TiFlash proxy [#856](https://github.com/pingcap/tics/pull/856)

+ TiCDC

    - replace table ID with table name in metrics [#695](https://github.com/pingcap/ticdc/pull/695)

+ BR

    - Support JSON log [#336](https://github.com/pingcap/br/issues/336)
    - Support starting pprof in-the-fly [#372](https://github.com/pingcap/br/pull/372)
    - Speed up DDL execution by send DDL concurrently [#377](https://github.com/pingcap/br/pull/377)

> + TiDB-Lightning

Reviewer: ```suggestion
+ TiDB Lightning
```

---
title: TiDB v4.0.3 Release Notes
category: Releases
aliases: ['/docs/dev/releases/v4.0.3/']
---

# TiDB v4.0.3 Release Notes

Release date: July 16, 2020

TiDB version: v4.0.3

## New Features

> 

Reviewer 2: + TiDB Dashboard
    - Display detailed TiDB Dashobard version information [#679](https://github.com/pingcap-incubator/tidb-dashboard/pull/679)
    - Show browser compatibility notice for unsupported browsers or outdated browsers [#654](https://github.com/pingcap-incubator/tidb-dashboard/pull/654)
    - Support searching in statement page [#658](https://github.com/pingcap-incubator/tidb-dashboard/pull/658)

    - Fix connection problem in some TLS scenarios [#660](https://github.com/pingcap-incubator/tidb-dashboard/pull/660)
    - Fix language dropdown not displayed correctly in some case [#677](https://github.com/pingcap-incubator/tidb-dashboard/pull/677)

+ TiFlash

    - Fix the issue that TiFlash crashes after renaming primary key column name. [#916](https://github.com/pingcap/tics/pull/916)
    - Fix the issue that concurrent learner read and remove region may lead to deadlocks. [#872](https://github.com/pingcap/tics/pull/872)

+ TiCDC

    - Fix the issue that TiCDC leaks memory in some cases [#704](https://github.com/pingcap/ticdc/pull/704)
    - Fix the issue that unquoted table name causes SQL syntax error [#676](https://github.com/pingcap/ticdc/pull/676)
    - Fix processor doesn't fully exit after `p.stop` is called [#693](https://github.com/pingcap/ticdc/pull/693)

+ BR

    - Fix the problem that backup time may be negative [#405](https://github.com/pingcap/br/pull/405)

+ Dumpling

    - Fix the issue that dumpling omit the NULL value when `--r` is specified [#119](https://github.com/pingcap/dumpling/pull/119)

>     - Fix the bug that flush tables may not work for tables to dump [#](https://github.com/pingcap/dumpling/pull/117)

Reviewer 3: ```suggestion
    - Fix the bug that flush tables may not work for tables to dump [#117](https://github.com/pingcap/dumpling/pull/117)
```

## Others

+ TiDB

    - Add system tables `tiflash_tables` and `tiflash_segments` [#18536](https://github.com/pingcap/tidb/pull/18536)
    - Executor: return the OOM error when indexHashJoin worker panics [#18527](https://github.com/pingcap/tidb/pull/18527)
    - Enable collect execution information by default [#18518](https://github.com/pingcap/tidb/pull/18518)
    - To speed up canceling a query [#18505](https://github.com/pingcap/tidb/pull/18505)
    - Change the default value of OOMAction to OOMActionCancel. [#18502](https://github.com/pingcap/tidb/pull/18502)
    - Add a header for tidb_decode_plan result. [#18501](https://github.com/pingcap/tidb/pull/18501)
    - Make the configuration checker compatible with older versions of the configuration [#18046](https://github.com/pingcap/tidb/pull/18046)
    - Insert negative primary key into auto_random table won't trigger rebase. [#17987](https://github.com/pingcap/tidb/pull/17987)

+ TiKV

    - Support dynamically changing the shared block cache's capacity [#8232](https://github.com/tikv/tikv/pull/8232)

>     - Fix an issue that read may get stale data during merging [#8113](https://github.com/tikv/tikv/pull/8113)

Reviewer: @uglyengineer Please help check the notes of this `Others` section belongs to which category. Thanks!

    - Add system tables `tiflash_tables` and `tiflash_segments` [#18536](https://github.com/pingcap/tidb/pull/18536)

+ TiKV

    - A new configuration `backup.num-threads` is introduced to control the backup thread pool size. [#8199](https://github.com/tikv/tikv/pull/8199)
    - Do not send store heartbeat when receiving snapshot [#8136](https://github.com/tikv/tikv/pull/8136)
    - Support dynamically changing the shared block cache's capacity [#8232](https://github.com/tikv/tikv/pull/8232)

+ PD

    - Support json formatted log [#2565](https://github.com/pingcap/pd/pull/2565)
    
+ Dashboard

    - Improved KeyViz bucket merge for cold logical ranges [#674](https://github.com/pingcap-incubator/tidb-dashboard/pull/674)
    - Renamed config item disable-telemetry to enable-telemetry to be more consistent [#684](https://github.com/pingcap-incubator/tidb-dashboard/pull/684)
    - Show progress bar when switching pages [#661](https://github.com/pingcap-incubator/tidb-dashboard/pull/661)
    - Slow log search now follows the same behaviour as log search when there are space separators [#682](https://github.com/pingcap-incubator/tidb-dashboard/pull/682)

+ TiFlash

>     - Change the unit of the panels of DDL jobs in Grafana to operations per minute [#923](https://github.com/pingcap/tics/pull/923)

Reviewer 4: ```suggestion
    - Change the unit of the panels of DDL jobs in Grafana to operations per minute [#923](https://github.com/pingcap/tics/pull/923)
    - Add a new dashboard in Grafana to show more metrics about TiFlah-Proxy [#923](https://github.com/pingcap/tics/pull/923)
```

    - Support specify output filename format [#122](https://github.com/pingcap/dumpling/pull/122)

## Bug Fixes

+ TiDB

    - Return an error instead of an empty set for IndexHashJoin when an error happens during execution. [#18586](https://github.com/pingcap/tidb/pull/18586)
    - Fix infinite panic when GRPC transportReader is broken [#18562](https://github.com/pingcap/tidb/pull/18562)
    - Fix the issue that green GC doesn't scan locks on offline stores which may cause data incompleteness. [#18550](https://github.com/pingcap/tidb/pull/18550)
    - Ban the TiFlash engine for a not-read-only statement. [#18534](https://github.com/pingcap/tidb/pull/18534)
    - Return the actual error message when a query connection panics. [#18500](https://github.com/pingcap/tidb/pull/18500)
    - DDL: fix admin repair table will reload fail on the other node [#18323](https://github.com/pingcap/tidb/pull/18323)
    - Fix read/write inconsistent result when meet lock that point to a primary key has be insert/delete in own txn [#18291](https://github.com/pingcap/tidb/pull/18291)
    - Make spilling disk work well. [#18288](https://github.com/pingcap/tidb/pull/18288)

>     - Fix a bug that replaces into a table with generated column fail [#17907](https://github.com/pingcap/tidb/pull/17907)

Reviewer 5: ```suggestion
    - Fix a bug that replaces into a table with generated column fail [#17907](https://github.com/pingcap/tidb/pull/17907)
    - Executor: return the OOM error when indexHashJoin worker panics [#18527](https://github.com/pingcap/tidb/pull/18527)
```

+ TiCDC

    - Fix the issue that TiCDC leaks memory in some cases [#704](https://github.com/pingcap/ticdc/pull/704)
    - Fix the issue that unquoted table name causes SQL syntax error [#676](https://github.com/pingcap/ticdc/pull/676)
    - Fix processor doesn't fully exit after `p.stop` is called [#693](https://github.com/pingcap/ticdc/pull/693)

+ BR

    - Fix the problem that backup time may be negative [#405](https://github.com/pingcap/br/pull/405)

+ Dumpling

    - Fix the issue that dumpling omit the NULL value when `--r` is specified [#119](https://github.com/pingcap/dumpling/pull/119)
    - Fix the bug that flush tables may not work for tables to dump [#](https://github.com/pingcap/dumpling/pull/117)

+ TiDB Lightning

    - Fix the issue that `--log-file` does not take effect [#345](https://github.com/pingcap/tidb-lightning/pull/345)

## Improvements

+ TiDB

    - Add `tidb_log_desensitization` global variable to control whether do desensitization when log query. [#18581](https://github.com/pingcap/tidb/pull/18581)
    - Active tidb_allow_batch_cop by default [#18552](https://github.com/pingcap/tidb/pull/18552)

>     - Fix issue of query `information_schema.columns` cost too many memory when there are lots of tables in TiDB. [#18436](https://github.com/pingcap/tidb/pull/18436)

Reviewer 5: ```suggestion
    - Fix issue of query `information_schema.columns` cost too many memory when there are lots of tables in TiDB. [#18436](https://github.com/pingcap/tidb/pull/18436)
    - Speed up canceling a query [#18505](https://github.com/pingcap/tidb/pull/18505)
    - Change the default value of OOMAction to OOMActionCancel. [#18502](https://github.com/pingcap/tidb/pull/18502)
    - Add a header for tidb_decode_plan result. [#18501](https://github.com/pingcap/tidb/pull/18501)
    - Make the configuration checker compatible with older versions of the configuration [#18046](https://github.com/pingcap/tidb/pull/18046)
    - Insert negative primary key into auto_random table won't trigger rebase. [#17987](https://github.com/pingcap/tidb/pull/17987)
    - Enable collect execution information by default [#18518](https://github.com/pingcap/tidb/pull/18518)
    - Add system tables `tiflash_tables` and `tiflash_segments` [#18536](https://github.com/pingcap/tidb/pull/18536)
```

    - Support starting pprof in-the-fly [#372](https://github.com/pingcap/br/pull/372)
    - Speed up DDL execution by send DDL concurrently [#377](https://github.com/pingcap/br/pull/377)

+ TiDB Lightning

    - [black-white-list] has been deprecated with a newer, easier-to-understand filter format [#332](https://github.com/pingcap/tidb-lightning/pull/332)

## Others

+ TiDB

    - Add system tables `tiflash_tables` and `tiflash_segments` [#18536](https://github.com/pingcap/tidb/pull/18536)
    - Executor: return the OOM error when indexHashJoin worker panics [#18527](https://github.com/pingcap/tidb/pull/18527)
    - Enable collect execution information by default [#18518](https://github.com/pingcap/tidb/pull/18518)
    - To speed up canceling a query [#18505](https://github.com/pingcap/tidb/pull/18505)
    - Change the default value of OOMAction to OOMActionCancel. [#18502](https://github.com/pingcap/tidb/pull/18502)
    - Add a header for tidb_decode_plan result. [#18501](https://github.com/pingcap/tidb/pull/18501)
    - Make the configuration checker compatible with older versions of the configuration [#18046](https://github.com/pingcap/tidb/pull/18046)

>     - Insert negative primary key into auto_random table won't trigger rebase. [#17987](https://github.com/pingcap/tidb/pull/17987)

Reviewer: @uglyengineer Please help confirm these notes belong to which category.

> ## Others
> 
> + TiDB
> 
>     - Add system tables `tiflash_tables` and `tiflash_segments` [#18536](https://github.com/pingcap/tidb/pull/18536)
>     - Executor: return the OOM error when indexHashJoin worker panics [#18527](https://github.com/pingcap/tidb/pull/18527)
>     - Enable collect execution information by default [#18518](https://github.com/pingcap/tidb/pull/18518)
>     - To speed up canceling a query [#18505](https://github.com/pingcap/tidb/pull/18505)
>     - Change the default value of OOMAction to OOMActionCancel. [#18502](https://github.com/pingcap/tidb/pull/18502)
>     - Add a header for tidb_decode_plan result. [#18501](https://github.com/pingcap/tidb/pull/18501)
>     - Make the configuration checker compatible with older versions of the configuration [#18046](https://github.com/pingcap/tidb/pull/18046)
>     - Insert negative primary key into auto_random table won't trigger rebase. [#17987](https://github.com/pingcap/tidb/pull/17987)

Reviewer 5: ```suggestion
```

---
title: TiDB v4.0.3 Release Notes
category: Releases
aliases: ['/docs/dev/releases/v4.0.3/']
---

# TiDB v4.0.3 Release Notes

Release date: July 16, 2020

TiDB version: v4.0.3

## New Features

+ TiFlash

>     - Implement file encryption in tiflash proxy. [#888](https://github.com/pingcap/tics/pull/888)

Reviewer 6: ```suggestion
    - Implement file encryption in TiFlash proxy. [#888](https://github.com/pingcap/tics/pull/888)
```

> 

Reviewer 3: ```suggestion

+ BR

    - Support compressing backup files using zstd, lz4 or snappy [#404](https://github.com/pingcap/br/pull/404)

+ TiCDC

    - Support to configure kafka-client-id in mq sink-uri [#706](https://github.com/pingcap/ticdc/pull/706)
    - Support updating changefeed config offline [#699](https://github.com/pingcap/ticdc/pull/699)
    - Support setting custom changefeed name [#727](https://github.com/pingcap/ticdc/pull/727)
    - Support TLS and MySQL SSL connection [#347](https://github.com/pingcap/ticdc/pull/347)
    - Support output changes in Avro format [#753](https://github.com/pingcap/ticdc/pull/753)
    - Support Apach Pulsar sink [#751](https://github.com/pingcap/ticdc/pull/751)

+ Dumpling

    - Support specialized csv separator and delimiter [#116](https://github.com/pingcap/dumpling/pull/116)
    - Support specify output filename format [#122](https://github.com/pingcap/dumpling/pull/122)

```

## Bug Fixes

+ TiDB

    - Return an error instead of an empty set for IndexHashJoin when an error happens during execution. [#18586](https://github.com/pingcap/tidb/pull/18586)
    - Fix infinite panic when GRPC transportReader is broken [#18562](https://github.com/pingcap/tidb/pull/18562)
    - Fix the issue that green GC doesn't scan locks on offline stores which may cause data incompleteness. [#18550](https://github.com/pingcap/tidb/pull/18550)
    - Ban the TiFlash engine for a not-read-only statement. [#18534](https://github.com/pingcap/tidb/pull/18534)
    - Return the actual error message when a query connection panics. [#18500](https://github.com/pingcap/tidb/pull/18500)

>     - Ddl: fix admin repair table will reload fail on the other node [#18323](https://github.com/pingcap/tidb/pull/18323)

Reviewer 5: ```suggestion
    - DDL: fix admin repair table will reload fail on the other node [#18323](https://github.com/pingcap/tidb/pull/18323)
```

>     - No release note, can use #18244's release note [#18291](https://github.com/pingcap/tidb/pull/18291)

Reviewer 5: ```suggestion
    - Fix read/write inconsistent result when meet lock that point to a primary key has be insert/delete in own txn [#18291](https://github.com/pingcap/tidb/pull/18291)
```

>     - [Bug-fix] Make spilling disk work well. [#18288](https://github.com/pingcap/tidb/pull/18288)

Reviewer 5: ```suggestion
    - Make spilling disk work well. [#18288](https://github.com/pingcap/tidb/pull/18288)
```

    - Fix the issue that caused deadlock when deleting scheduler [#2637](https://github.com/pingcap/pd/pull/2637)
    - Fix the bug that balance-leader-scheduler won't consider placement rule when it enabled. [#2636](https://github.com/pingcap/pd/pull/2636)
    - Fixes the issue that sometimes service safepoint can't be set properly, which may make BR and dumpling fail. [#2635](https://github.com/pingcap/pd/pull/2635)
    - Fix dst store filter in hot region scheduler [#2627](https://github.com/pingcap/pd/pull/2627)
    - Fix the issue that the TSO request may take too long when PD leader switches [#2622](https://github.com/pingcap/pd/pull/2622)
    - Fix the problem of stale scheduler after leader change [#2608](https://github.com/pingcap/pd/pull/2608)
    - Fix the issue that when enable placement rules, sometimes replicas of region may not be able to adjust to the best location [#2605](https://github.com/pingcap/pd/pull/2605)
    - Fix the issue that the deploy path of the store is not updated if the deploy directory is moved [#2600](https://github.com/pingcap/pd/pull/2600)
    - Prevent store limit from changing to zero [#2588](https://github.com/pingcap/pd/pull/2588)

> 

Reviewer 2: ```suggestion
Dashboard
- Fix TiDB connection error when TiDB is scaled-in https://github.com/pingcap-incubator/tidb-dashboard/pull/689
- Fix TiFlash instances not displayed in log searching page https://github.com/pingcap-incubator/tidb-dashboard/pull/680
- Fix metrics selection reset after refresh in overview page https://github.com/pingcap-incubator/tidb-dashboard/pull/663
- Fix connection problem in some TLS scenarios https://github.com/pingcap-incubator/tidb-dashboard/pull/660
- Fix language dropdown not displayed correctly in some case https://github.com/pingcap-incubator/tidb-dashboard/pull/677
```

+ TiFlash

>     - Fix the issue that TiFlash crash after renaming primary key column name. [#916](https://github.com/pingcap/tics/pull/916)

Reviewer 6: ```suggestion
    - Fix the issue that TiFlash crashes after renaming primary key column name. [#916](https://github.com/pingcap/tics/pull/916)
```

>     - Fix the issue that concurrency learner read and remove region may meet deadlocks. [#872](https://github.com/pingcap/tics/pull/872)

Reviewer 6: ```suggestion
    - Fix the issue that concurrent learner read and remove region may lead to deadlocks. [#872](https://github.com/pingcap/tics/pull/872)
```

> 

Reviewer 3: ```suggestion

+ TiCDC

    - Fix the issue that TiCDC leaks memory in some cases [#704](https://github.com/pingcap/ticdc/pull/704)
    - Fix the issue that unquoted table name causes SQL syntax error [#676](https://github.com/pingcap/ticdc/pull/676)
    - Fix processor doesn't fully exit after `p.stop` is called [#693](https://github.com/pingcap/ticdc/pull/693)

+ BR

    - Fix the problem that backup time may be negative [#405](https://github.com/pingcap/br/pull/405)

+ Dumpling

    - Fix the issue that dumpling omit the NULL value when `--r` is specified [#119](https://github.com/pingcap/dumpling/pull/119)
    - Fix the bug that flush tables may not work for tables to dump [#](https://github.com/pingcap/dumpling/pull/117)

+ TiDB-Lightning

    - Fix the issue that `--log-file` does not take effect [#345](https://github.com/pingcap/tidb-lightning/pull/345)

```

## Improvements

+ TiDB

    - Add `tidb_log_desensitization` global variable to control whether do desensitization when log query. [#18581](https://github.com/pingcap/tidb/pull/18581)
    - Active tidb_allow_batch_cop by default. [#18552](https://github.com/pingcap/tidb/pull/18552)
    - Fix issue of query `information_schema.columns` cost too many memory when there are lots of tables in TiDB. [#18436](https://github.com/pingcap/tidb/pull/18436)

+ TiKV

    - A new configuration `backup.num-threads` is introduced to control the backup thread pool size. [#8199](https://github.com/tikv/tikv/pull/8199)
    - No release notes. [#8190](https://github.com/tikv/tikv/pull/8190)

>     - Do not send store heartbeat when snapshot received [#8136](https://github.com/tikv/tikv/pull/8136)

Reviewer 7: ```suggestion
    - Do not send store heartbeat when receiving snapshot [#8136](https://github.com/tikv/tikv/pull/8136)
```

+ PD

    - Support json formatted log [#2565](https://github.com/pingcap/pd/pull/2565)

> 

Reviewer 2: ```suggestion
Dashboard
- Improved KeyViz bucket merge for cold logical ranges https://github.com/pingcap-incubator/tidb-dashboard/pull/674
- Renamed config item disable-telemetry to enable-telemetry to be more consistent https://github.com/pingcap-incubator/tidb-dashboard/pull/684
- Display detailed TiDB Dashobard version information https://github.com/pingcap-incubator/tidb-dashboard/pull/679
- Show progress bar when switching pages https://github.com/pingcap-incubator/tidb-dashboard/pull/661
- Show browser compatibility notice for unsupported browsers or outdated browsers https://github.com/pingcap-incubator/tidb-dashboard/pull/654
- Support searching in statement page https://github.com/pingcap-incubator/tidb-dashboard/pull/658
- Slow log search now follows the same behaviour as log search when there are space separators https://github.com/pingcap-incubator/tidb-dashboard/pull/682
```

+ TiFlash

    - Change the unit of the panels of DDL jobs in Grafana to operations per minute [#923](https://github.com/pingcap/tics/pull/923)

>     - Reduce consumption of iops in raft module [#856](https://github.com/pingcap/tics/pull/856)

Reviewer 6: ```suggestion
    - Reduce IOPS in TiFlash proxy [#856](https://github.com/pingcap/tics/pull/856)
```

> 

Reviewer 3: ```suggestion

+ TiCDC

    - replace table ID with table name in metrics [#695](https://github.com/pingcap/ticdc/pull/695)

+ BR

    - Support JSON log [#336](https://github.com/pingcap/br/issues/336)
    - Support starting pprof in-the-fly [#372](https://github.com/pingcap/br/pull/372)
    - Speed up DDL execution by send DDL concurrently [#377](https://github.com/pingcap/br/pull/377)

+ TiDB-Lightning

    - [black-white-list] has been deprecated with a newer, easier-to-understand filter format [#332](https://github.com/pingcap/tidb-lightning/pull/332)

```

## Others

+ TiDB

    - Add system tables `tiflash_tables` and `tiflash_segments` [#18536](https://github.com/pingcap/tidb/pull/18536)
    - Executor: return the OOM error when indexHashJoin worker panics [#18527](https://github.com/pingcap/tidb/pull/18527)
    - Enable collect execution information by default [#18518](https://github.com/pingcap/tidb/pull/18518)
    - To speed up canceling a query [#18505](https://github.com/pingcap/tidb/pull/18505)
    - Change the default value of OOMAction to OOMActionCancel. [#18502](https://github.com/pingcap/tidb/pull/18502)
    - Add a header for tidb_decode_plan result. [#18501](https://github.com/pingcap/tidb/pull/18501)
    - Make the configuration checker compatible with older versions of the configuration [#18046](https://github.com/pingcap/tidb/pull/18046)
    - Insert negative primary key into auto_random table won't trigger rebase. [#17987](https://github.com/pingcap/tidb/pull/17987)

+ TiKV

>     - Support dynamically changing the shared block cache's capacity [#8232](https://github.com/tikv/tikv/pull/8232)

Reviewer 7: This is better to be an improvement

>     - Fix `test_security_status_service_with_cn` unit test [#8181](https://github.com/tikv/tikv/pull/8181)

Reviewer 7: ```suggestion
```
Delete this

>     - Fix an issue that read may get stale data during merging [#8113](https://github.com/tikv/tikv/pull/8113)

Reviewer 7: This should be a bugfix

>     - Copr: support collation aware min/max aggregation functions [#8108](https://github.com/tikv/tikv/pull/8108)

Reviewer 7: ```suggestion
    - Fix the issue that collation does not work on min/max function when aggregation is pushed down to tikv  [#8108](https://github.com/tikv/tikv/pull/8108)
```
This should be a bugfix

> + Tools
> 
>     - BR
> 
>         * Improve parallel processing power when backing up a large database with many regions. [#410](https://github.com/pingcap/br/pull/410)

Reviewer 3: ```suggestion
```
